elastic/beats
# [aws] Set region in config from GetBucketLocation API call

## Proposed commit message

This PR sets the region of the S3 bucket as the region in AWS config to make API calls. 

## Checklist




## How to test this PR locally

Without this change, run Filebeat `aws-s3` input with this config and you will see error:
```
s3 GetObject failed: operation error S3: GetObject, https response error StatusCode: 301, RequestID: 01H7T2XWRZ6N44MW, HostID: WqYAZM+diCdCnBwj2i1tAzchSXBZql8SdIfQ0ZUCKHoR4YbMiBGD10duVYqIQ97pCg/j8MmJTgo=, api error PermanentRedirect: The bucket you are attempting to access must be addressed using the specified endpoint. Please send all future requests to this endpoint.
```
config:
```
filebeat.inputs:
- type: aws-s3
  enabled: true
  credential_profile_name: elastic-observability
  number_of_workers: 5
  bucket_arn: 'arn:aws:s3:::tkravchenko-test1'
```
The workaround is to add `default_region: us-east-2` in the config.

With this change, Filebeat is able to get the bucket location/region from API call and assign it to the region config.
